x-pack/plugins/security_solution/public/common/containers/source/use_data_view.tsx

```diff
@@ -51,20 +51,58 @@ export const getDataViewStateFromIndexFields = memoizeOne(
     type DangerCastForMutation = Record<string, {}>;
     if (fields == null) {
       return { browserFields: {} };
-    } else {
-      const browserFields: BrowserFields = {};
-      for (const [name, field] of Object.entries(fields)) {
+    }
+
+    const browserFields: BrowserFields = {};
+    for (const [name, field] of Object.entries(fields)) {
+      const category = getCategory(name);
+      if (browserFields[category] == null) {
+        (browserFields as DangerCastForMutation)[category] = { fields: {} };
+      }
+      const categoryFields = browserFields[category].fields;
+      if (categoryFields) {
+        categoryFields[name] = field;
+      }
+    }
+    return { browserFields: browserFields as DangerCastForBrowserFieldsMutation };
+  },
+  (newArgs, lastArgs) => newArgs[0] === lastArgs[0] && newArgs[1]?.length === lastArgs[1]?.length
+);
+
+// Categories fields in smaller chunks to avoid blocking the event loop
+export const getDataViewStateFromIndexFieldsAsync = memoizeOne(
```

Comment:
It is strange that there is no test available for this. If you have time and can add some tests, that will be awesome but not a blocker from my side.

Thank you.